src/abstractdataframe/abstractdataframe.jl

```diff
@@ -1450,6 +1450,146 @@ julia> unique!(df)  # modifies df
 """
 (unique, unique!)
 
+"""
+    completecombinations(df::AbstractDataFrame, indexcols;
+                         allowduplicates::Bool=false,
+                         fill=missing)
+
+Generate all combinations of levels of column(s) `indexcols` in data frame `df`.
+Levels and their order are determined by the `levels` function
+(i.e. unique values sorted lexicographically by default, or a custom set
+of levels for e.g. `CategoricalArray` columns), in addition to `missing` if present.
+
```

Comment:
This still applies, right?
```suggestion

All columns from `df` are added at the end of the returned data frame.
For combinations of `indexcols` not present in `df` these columns are
filled with the `fill` value (`missing` by default).

```

Reply:
I have corrected your suggestion this. Now we keep the columns in the order of source data frame (but indeed I have accidentally deleted the `fill` part).